src/components/OnchainName.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useMemo } from 'react';
+import { getSlicedAddress } from '../core/address';
+import { useEnsName } from '../hooks/useEnsName';
+import type { Address } from 'viem';
+
+type OnchainNameProps = {
+  address?: Address;
+  className?: string;
+  sliced?: boolean;
+  props?: React.HTMLAttributes<HTMLSpanElement>;
+};
+
+/**
+ * OnchainName is a React component that renders the user name from an Ethereum address.
+ * It displays the ENS name if available; otherwise, it shows either a sliced version of the address
+ * or the full address, based on the 'sliced' prop. By default, 'sliced' is set to true.
+ *
+ * @param {Address} address - Ethereum address to be displayed.
+ * @param {string} [className] - Optional CSS class for custom styling.
+ * @param {boolean} [sliced=true] - Determines if the address should be sliced when no ENS name is available.
+ * @param {React.HTMLAttributes<HTMLSpanElement>} [props] - Additional HTML attributes for the span element.
+ */
+export function OnchainName({ address, className, sliced = true, props }: OnchainNameProps) {
+  const { ensName } = useEnsName(address);
+
+  const normilizedAddress = useMemo(() => {
```

Comment:
Add a comment here why we have `useMemo`

Reply:
Also should we avoid have CPU execution for the `normilizedAddress` if in case of ensName we will use that anyway?

Reply:
I think the idea here is to always display 
```suggestion
  const normalizedAddress = useMemo(() => {
```

Reply:
Also, I think the idea here is that the input address will display until ens name resolves. I don't think that's unreasonable, but maybe we'd prefer an explicit loading state?

Reply:
Great call. Yes, we need loading state here. Wondering how the UX would look like when you have a list of users in the UX, will things shift and change like crazy. User might opt-in to display loading state. 

Need some UX thinking here.